src/patch.dev.js

```diff
@@ -77,26 +77,42 @@ if (React.createElement.isPatchedByReactHotLoader) {
 // https://github.com/gaearon/react-hot-loader/issues/249
 function isReactRouterish(type) {
   return type && (
-    type.displayName === 'Route' ||
-    type.name === 'Route' || // In case Ryan and Michael embrace ES6 classes
-    type.displayName === 'IndexRoute' ||
-    type.name === 'IndexRoute'
+    type.displayName === 'Router' ||
+    type.name === 'Router' // In case Ryan and Michael embrace ES6 classes
   );
 }
 
+function forceUpdateComponentOfRouteAndChildRoutes(route) {
+  // TODO: handle multiple components/route
+  // TODO: check whether it is possible to also handle the `getComponent` case here
+  if (route.component && typeof route.component === 'function') {
+    // Side effect 😱
+    // Force proxies to update since React Router ignores new props.
+    resolveType(route.component);
+  }
+
+  // Child routes will be in `props.children` when defining routes using react
+  // elements and in `routes` or `childRoutes` when using plain routes
+  var childRoutes = []
+    .concat(
+      route.routes,
+      route.childRoutes,
+      route.props && route.props.children)
+    .filter(Boolean);
+
+  for (var i = 0; i < childRoutes.length; i++) {
+    var childRoute = childRoutes[_key];
```

Comment:
Yeah, nice catch. It seems like it worked because babel were transpiling it to _key anyways.